core/src/main/csharp/ch/cyberduck/core/CredentialManagerPasswordStore.cs

```diff
@@ -208,26 +219,78 @@ public override void save(Host bookmark)
             }
         }
 
-        private static Uri ToUri(Host bookmark)
+        private static Uri ToUri(Host bookmark) => ToUri(bookmark, true, out _);
+
+        private static Uri ToUri(Host bookmark, bool withOAuth, out bool isOAuth)
         {
             var protocol = bookmark.getProtocol();
             var credentials = bookmark.getCredentials();
-
             var targetBuilder = new UriBuilder(PreferencesFactory.get().getProperty("application.container.name"), string.Empty);
             var pathBuilder = new StringBuilder();
-            pathBuilder.Append(protocol.getIdentifier());
-            if (protocol.isHostnameConfigurable() || !(protocol.isTokenConfigurable() || protocol.isOAuthConfigurable()))
+
+            int? port;
+            string hostname = default;
+            string username = credentials.getUsername();
+
+            if (withOAuth && protocol.isOAuthConfigurable())
             {
-                pathBuilder.Append(":" + bookmark.getHostname());
-                if (protocol.isPortConfigurable() && !Equals(protocol.getDefaultPort(), bookmark.getPort()))
+                isOAuth = true;
+                OAuthPrefixService oAuthPrefix = new OAuthPrefixServiceFactory().create(bookmark);
+                hostname = oAuthPrefix.getIdentifier();
```

Comment:
@AliveDevil looks dodgy to assign `identifier` to `hostname`

Reply:
Well, that's what I called the field, could change that to `identifier`, which holds either `OAuthPrefix.getIdentifier()` or `bookmark.getHostname()` (in non-OAuth path).
Which is then built to `duck:identifier:port?user=` (OAuth) or `duck:fqdn:port?user=` (Default).

Reply:
This causes a regression as previously entries were saved using `bookmark.getProtocol().getOAuthTokenUrl()` when available.

Reply:
Only using the protocol identifier and no token URL will cause entries to be saved with the same key for bookmarks using same protocol but different OAuth endpoint.

Reply:
@AliveDevil The previous default was `duck:identifier:fqdn:port?user= `, not `duck:fqdn:port?user= `?